cmd/object-handlers.go

```diff
@@ -813,8 +813,10 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 		writeErrorResponse(w, toAPIErrorCode(ctx, err), r.URL, guessIsBrowserReq(r))
 		return
 	}
+
 	rawReader := srcInfo.Reader
-	pReader := NewPutObjReader(srcInfo.Reader, srcInfo.Reader, nil)
+	pReader := NewPutObjReader(rawReader, nil, nil)
```

Comment:
I think second arg also needs to be `rawReader` - but @poornas can confirm.

Reply:
no it is not @donatello - there is no reason to provide second reader without encryption key.